src/mds/mdstypes.h

```diff
@@ -1020,4 +1020,22 @@ inline bool operator==(const MDSCacheObjectInfo& l, const MDSCacheObjectInfo& r)
 }
 WRITE_CLASS_ENCODER(MDSCacheObjectInfo)
 
+struct EstimatedReplayTime {
+  EstimatedReplayTime(double percent_complete, std::chrono::seconds estimated_time,
+		      std::chrono::seconds elapsed_time)
+    : percent_complete(percent_complete),
+      estimated_time(estimated_time),
+      elapsed_time(elapsed_time) {
+  }
+
+  double percent_complete;
+  std::chrono::seconds estimated_time;
+  std::chrono::seconds elapsed_time;
+};
+
+inline std::ostream& operator<<(std::ostream& out, const EstimatedReplayTime &estimate) {
+  return out << "replay: " << estimate.percent_complete << "\% complete - elapsed time: "
+	     << estimate.elapsed_time << "s, estimated time remaining " << estimate.estimated_time << "s." ;
+}
```

Comment:
Suggest adding a print method, using 92df0fcc649345f438d0d7cc6fab658c02060f07. It's cleaner IMO.